web3/utils/datatypes.py

```diff
@@ -0,0 +1,45 @@
+from functools import (
+    partial,
+)
+from eth_utils import (
+    to_dict,
+)
+
+from web3.utils.normalizers import (
+    normalize_abi,
+    normalize_address,
+    normalize_bytecode,
+)
+
+
+class PropertyCheckingFactory(type):
+    ens = None
+
+    @to_dict
+    def normalize_properties(attributes, values, ens):
```

Comment:
I don't love that this is included in the `PropertyCheckingFactory` since it's specific to the `Contract` class.  Could we change this so that `__new__` accepted an *optional* `normalizers` argument?